auditlog/models/rule.py

```diff
@@ -237,28 +237,31 @@ def _make_read(self):
         self.ensure_one()
         log_type = self.log_type
 
-        def read(self, *args, **kwargs):
-            result = read.origin(self, *args, **kwargs)
+        @api.multi
+        def read(self, fields=None, load='_classic_read'):
+            result = read.origin(self, fields, load)
             # Sometimes the result is not a list but a dictionary
             # Also, we can not modify the current result as it will break calls
             result2 = result
             if not isinstance(result2, list):
                 result2 = [result]
             read_values = dict((d['id'], d) for d in result2)
             # Old API
-            if args and isinstance(args[0], sql_db.Cursor):
-                cr, uid, ids = args[0], args[1], args[2]
+            if self and isinstance(self.env.cr, sql_db.Cursor):
+                cr, uid, ids = self.env.cr, self.env.uid, self.ids
                 if isinstance(ids, (int, long)):
                     ids = [ids]
                 # If the call came from auditlog itself, skip logging:
                 # avoid logs on `read` produced by auditlog during internal
                 # processing: read data of relevant records, 'ir.model',
                 # 'ir.model.fields'... (no interest in logging such operations)
-                if kwargs.get('context', {}).get('auditlog_disabled'):
+                if self.env.context.get('context', {}).\
+                   get('auditlog_disabled'):
                     return result
                 env = api.Environment(cr, uid, {'auditlog_disabled': True})
                 rule_model = env['auditlog.rule']
-                rule_model.sudo().create_logs(
+                rule_model.sudo().with_context({'auditlog_disabled': True}).\
+                    create_logs(
```

Comment:
Why changing this? the environment has already the `auditlog_disabled` set to True (two lines above).